File: Runtime/TransitionGroupCollection.vb
'************************************************************************************
' ST-Sim: A .NET class library for developing state-and-transition simulation models.
'
' Copyright © 2007-2017 Apex Resource Management Solution Ltd. (ApexRMS). All rights reserved.
'
'************************************************************************************

Imports System.Collections.ObjectModel

''' <summary>
''' Collection of TransitionGroups keyed by TransitionGroupID
''' </summary>
Public Class TransitionGroupCollection
    Inherits KeyedCollection(Of Integer, TransitionGroup)

    ''' <summary>
    ''' Constructor
    ''' </summary>
    ''' <remarks>
    ''' Prevent public constructor
    ''' </remarks>
    Friend Sub New()
        Return
    End Sub

    Protected Overrides Function GetKeyForItem(ByVal item As TransitionGroup) As Integer
        Return item.TransitionGroupId
    End Function

End Class

File: Runtime/TransitionGroupCollection.cs
// ST-Sim: A SyncroSim Module for the ST-Sim State-and-Transition Model.
// Copyright © 2007-2018 Apex Resource Management Solution Ltd. (ApexRMS). All rights reserved.

using System.Collections.ObjectModel;

namespace SyncroSim.STSim
{
    public class TransitionGroupCollection : KeyedCollection<int, TransitionGroup>
    {
        internal TransitionGroupCollection()
        {
            return;
        }

        protected override int GetKeyForItem(TransitionGroup item)
        {
            return item.TransitionGroupId;
        }
    }
}
